test-new/e2e/GuardianVerifier.t.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {ModuleKitHelpers, AccountInstance} from "modulekit/ModuleKit.sol";
+import {MODULE_TYPE_EXECUTOR} from "modulekit/accounts/common/interfaces/IERC7579Module.sol";
+import {EmailAuthMsg} from "@zk-email/ether-email-auth-contracts/src/EmailAuth.sol";
+import {EmailAccountRecovery} from "@zk-email/ether-email-auth-contracts/src/EmailAccountRecovery.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+import {IEmailRecoveryManager} from "src/interfaces/IEmailRecoveryManager.sol";
+import {IGuardianManager} from "src/interfaces/IGuardianManager.sol";
+import {GuardianStorage, GuardianStatus} from "src/libraries/EnumerableGuardianMap.sol";
+
+import {CommandHandlerType, GuardianType} from "./GuardianVerifierBase.t.sol";
+import {OwnableValidatorRecovery_AbstractedRecoveryModule_Base} from "./GuardianVerifierBase.t.sol";
+
+/**
+ * Test the abstracted recovery module with email based guardians
+ */
+contract OwnableValidatorRecovery_AbstractedRecoveryModule_Test is
+    OwnableValidatorRecovery_AbstractedRecoveryModule_Base
+{
+    using ModuleKitHelpers for *;
+    using Strings for uint256;
+
+    function setUp() public override {
+        super.setUp();
+    }
+
+    // Helper function
+    function executeRecoveryFlowForAccount(
+        address account,
+        address[] memory guardians,
+        bytes32 recoveryDataHash,
+        bytes memory recoveryData
+    ) internal {
+        acceptGuardian(
+            GuardianType.EmailGuardian,
+            emailGuardianVerifierImplementation,
+            account,
+            guardians[0],
+            emailRecoveryModuleAddress,
+            accountSalt1,
+            emailGuardianVerifierInitData
+        );
+        acceptGuardian(
```

Comment:
Looks great. For completeness, could we also use the jwt guardian in this test? Could just be for acceptance flow for simplicity

Reply:
The only issue is not having a hosted prover, If it's available or you have old proofs that are valid, then maybe I can give it a shot.

A local prover is not an option for me, it's consuming extreme resources.